DuckDuckGo/Tab/UserScripts/Subscription/SubscriptionPagesUseSubscriptionFeature.swift

```diff
@@ -561,4 +573,21 @@ private extension SubscriptionPagesUseSubscriptionFeature {
             freemiumDBPUserStateManager.upgradeToSubscriptionTimestamp = Date()
         }
     }
+
+    /// Sets the origin for attribution if the user has started their first Freemium PIR scan
+    ///
+    /// This method checks whether the user has started their first Freemium PIR scan.
+    /// If they have, the method sets the subscription success tracking origin to `"funnel_pro_mac_freemium"` and returns `true`.
+    ///
+    /// - Returns:
+    ///   - `true` if the origin is set because the user has started their first Freemim PIR scan.
+    ///   - `false` if a first scan has not been started and the origin is not set.
+    func setFreemiumOriginIfScanPerformed() -> Bool {
+        let origin = "funnel_pro_mac_freemium"
```

Comment:
Do you think there would be a better place to define it as a const? We have `PrivacyProSubscriptionAttributionPixelHandler` so maybe there, wdyt?

Reply:
Thanks @miasma13, I’ve extracted the string to a constant in `PrivacyProSubscriptionAttributionPixelHandler` as suggested.